setup.py

```diff
@@ -44,21 +44,12 @@ def list_modules(dirname):
 
         return module_names
 
-    ext_modules = [
-        Extension('falcon.' + ext, [path.join('falcon', ext + '.py')])
-        for ext in list_modules(path.join(MYDIR, 'falcon'))]
-
-    ext_modules += [
-        Extension('falcon.util.' + ext,
-                  [path.join('falcon', 'util', ext + '.py')])
-
-        for ext in list_modules(path.join(MYDIR, 'falcon', 'util'))]
-
-    ext_modules += [
-        Extension('falcon.routing.' + ext,
-                  [path.join('falcon', 'routing', ext + '.py')])
+    dirs = ['falcon', 'falcon.util', 'falcon.routing']
+    ext_modules = []
 
-        for ext in list_modules(path.join(MYDIR, 'falcon', 'routing'))]
+    for each in dirs:
+        ext_modules += [Extension(each + ext, [path.join(each, ext + '.py')])
+                            for ext in list_modules(path.join(MYDIR, each))]
```

Comment:
Similar to the line above, `each` here is a dotted module name, and so would need to be split first.